### PR TITLE
Local filesystem storage

### PR DESCRIPTION
**Pull request by piotr-iohk**
Following with the changes from #9, this pr introduces local filesystem storage implementation in `delegation_backend` package.

### Overview

 - uptime-service-backend operator may use local filesystem as storage backend by adding to the `app_config.json`:
```
  "filesystem": {
    "path": "/path/to/local/dir"
  }
```
 - as implemented in #9 there can be only one storage backend, after this pr two are supported: `aws`, `filesystem`

### Testing

 - there was an integration test added that is testing if blocks and submissions are stored on local filesystem in case if `app_config.json` contains local filesystem config.

### CODE REVIEW

File: src/integration_tests/constants.go [hunk 1]
```diff
@@ -12,17 +14,48 @@ const (
...
+	var gpgFiles []string
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), suffix) {
+			absolutePath := dir + "/" + f.Name()
+			gpgFiles = append(gpgFiles, absolutePath)
+		}
+	}
+
+	return gpgFiles, nil
+}
```
Comment (smorci) [inline src/integration_tests/constants.go]: The variable in this function is named `gpgFiles` but is also used to get `json` files. A bit confusing. 
```suggestion
	var result []string
	for _, f := range files {
		if strings.HasSuffix(f.Name(), suffix) {
			absolutePath := dir + "/" + f.Name()
			result = append(result, absolutePath)
		}
	}

	return result, nil
}
```

File: src/integration_tests/integration_test.go
Comment (smorci) [inline src/integration_tests/integration_test.go:1]: Do we need two functions for this? Are these features mutually exclusive? Can't we test S3 and LocalFilesystem with the same minimina deployment? Seems rather wasteful to sync two minimina networks. Or are these done at least in parallel?

**Review by smorci — CHANGES_REQUESTED** [review]
Nothing major, just some minor comments.

**Review by Sventimir — COMMENTED** [review]
Looks okay to me.

File: src/integration_tests/integration_test.go
Comment (piotr-iohk) [reply src/integration_tests/integration_test.go:1]: Yes, atm we can have either `aws` or `filesystem` storage backend configured. I was going for a self contained e2e test for each config, but now that you mention this I guess we could swap the config and restart uptime-service-backend, while the network is running... Both tests run sequentially and take ~10m to complete, so it is not horrible... however let me check if swapping configs in runtime will cause them to run faster.

Atm these tests cannot run in parallel as there might be a port conflict when spinning two minimina networks next to each other (there is this item in minimina backlog to address this https://github.com/MinaFoundation/minimina/issues/37)

Comment (piotr-iohk) [reply src/integration_tests/integration_test.go:1]: Seems like merging these tests into one and making them use the same network results in much faster execution time :+1: 

```
--- PASS: TestIntegration_BP_Uptime_Storage (263.30s)
PASS
ok      block_producers_uptime/integration_tests        263.307s
```

Reworked.

**Review by smorci — APPROVED** [review]
Looks good!

**Pull request merged**